paparazzi/src/main/java/app/cash/paparazzi/internal/resources/PseudolocaleGenerator.kt

```diff
@@ -0,0 +1,144 @@
+package app.cash.paparazzi.internal.resources
+
+import app.cash.paparazzi.internal.resources.base.BasicArrayResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicPluralsResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicTextValueResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicValueResourceItem
+import app.cash.paparazzi.internal.resources.base.BasicValueResourceItemBase
+import com.android.ide.common.rendering.api.ResourceNamespace
+import com.android.ide.common.resources.ResourceValueMap
+import com.android.ide.common.resources.configuration.FolderConfiguration
+import com.android.ide.common.resources.configuration.LocaleQualifier
+import com.android.resources.ResourceType
+import com.google.common.collect.Table
+
+private const val DO_NOT_TRANSLATE = "donottranslate"
+
+private val configCache = hashMapOf<Pair<ResourceSourceFile, LocaleQualifier>, RepositoryConfiguration>()
```

Comment:
how beneficial is caching here?  i.e., how "heavy" or slow is it to calculate these extra resources at runtime?

also, will this translate framework resources as well?  if so, that alone might justify the caching.

would be curious if we also had some anecdotal measurements on how much the heap space grows as a result of this caching.  IIRC we saved about 1GB by migrating from legacy resource rewriting to current, is that correct?  If this new cache is in the order of MBs, then sgtm!

Reply:
Resource rewriting saved `50MB` actually. 

And yes the cache is for framework resources as well. From my test, the framework resources memory use is `~2.8MB` without cache and reduced to `~1.7MB` with cache

Reply:
The cache is about saving memory use. Without cache, we have to create a new `RepositoryConfiguration` every time when creating a new pseudo item. With cache, pseudo items can refer to cached `RepositoryConfiguration` for same source file. That’s how it saves memory